proposal/package-importer.md

```diff
@@ -0,0 +1,261 @@
+This proposal defines a new importer that implementations should make available.
+It will be disabled by default. The loader will handle URLs:
+
+- with the scheme `pkg`
+- followed by `:`
+- followed by a package name
+- optionally followed by a path, with path segments separated with a forward
+  slash.
+
+### Resolving a `pkg:` URL
+
+The `pkg importer` will provide a method to canonicalize a `pkg:` URL, and
+extend the implementation's existing File Importer.
+
+### Platform-Specific Semantics
+
+#### Dart
+
+Given `url` with the format `pkg:.*`:
+
+- Let `path` be `url` without the `pkg` scheme
+- Use [package-config] to resolve `path`
+
+[package-config]: https://pub.dev/packages/package_config
+
+#### Node
+
+Given `url` with the format `pkg:.*`:
+
+- Let `fullPath` be `url` without the `pkg` scheme
+- Let `resolved` be the result of using [resolve.exports] to resolve `fullPath`
+  with the `sass` condition set
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+- Let `resolved` be the result of using [resolve.exports] to resolve `fullPath`
+  with the `style` condition set.
+  - If `resolved` has the scheme `file:` and an extension of `css`, return it.
+- Let `packageName` be the package identifier, and `subPath` be the path without
+  the package identifier.
+- If `subPath` is empty, return result of [resolving `pkg` root].
+- Otherwise, return result of [resolving `pkg` subpath].
+
+[resolve.exports]: https://github.com/lukeed/resolve.exports
+[resolving pkg root]: #resolving-pkg-root
+[resolving pkg subpath]: #resolving-pkg-subpath
+
+#### Resolving `pkg` Root
+
+- Let `packagePath` be the file path to the package root.
+- Let `sassValue` be the value of `sass` in the `package.json` at the pkg root.
+- If `sassValue` is a relative path with an extension of `sass`, `scss` or
+  `css`, return the `packagePath` appended with `sassValue`.
+- Let `styleValue` be the value of `style` in the `package.json` at the pkg
+  root.
+- If `styleValue` is a relative path with an extension of `css`, return the
+  `packagePath` appended with `styleValue`.
+- Otherwise return the result of [resolving a file url] with `packagePath`.
+
+[resolving a file url]: ../spec/modules.md#resolving-a-file-url
+
+#### Resolving `pkg` Subpath
+
+- Let `packagePath` be the file path to the package root.
+- Let `fullPath` be `subpath` resolved relative to `packagePath`.
+- Return the result of [resolving a file url] with `fullPath`.
+
+#### Resolution Order
+
+This algorithm resolves in the following order:
+
+1. `sass` condition in package.json `exports`
+1. `style` condition in package.json `exports`
+1. If no subpath, then find root export:
+  1. `sass` key at package.json root
+  1. `style` key at package.json root
+  1. `index` file at package root, resolved for file extensions and partials
+1. If there is a subpath, resolve that path relative to the package root, and
+   resolve for file extensions and partials
```

Comment:
Style nit: even though Markdown will automatically count up ordered lists, it's still nice to count them up manually for people who are reading the plain text.